api/python/cellxgene_census/src/cellxgene_census/_get_anndata.py

```diff
@@ -75,14 +89,46 @@ def get_anndata(
     exp = _get_experiment(census, organism)
     obs_coords = (slice(None),) if obs_coords is None else (obs_coords,)
     var_coords = (slice(None),) if var_coords is None else (var_coords,)
+
     with exp.axis_query(
         measurement_name,
         obs_query=soma.AxisQuery(value_filter=obs_value_filter, coords=obs_coords),
         var_query=soma.AxisQuery(value_filter=var_value_filter, coords=var_coords),
     ) as query:
-        return query.to_anndata(
+        adata = query.to_anndata(
             X_name=X_name,
             column_names=column_names,
             X_layers=X_layers,
             obsm_layers=obsm_layers,
         )
+
+        # If add_obs_embeddings or add_var_embeddings are defined, inject them in the appropriate slot
+        if add_obs_embeddings or add_var_embeddings:
+            from .experimental._embedding import _get_embedding, get_embedding_metadata_by_name
+
+            census_version = _extract_census_version(census)
+            experiment_name = _get_experiment_name(organism)
+            census_directory = get_census_version_directory()
+
+            if add_obs_embeddings:
+                if obsm_layers and [x for x in add_obs_embeddings if x in obsm_layers]:
```

Comment:
this error check (name collision) is missing from the var axis.  Seems like you need it for both

Reply:
I removed `varm_layers` from the arguments, as you suggested, since it is not in the current API. The only way you can request a varm is through `add_varm_embeddings`

Reply:
Right - gotcha!

Reply:
So, I'm thinking about this previous conversation a bit more.  Coming around to a slightly different perspective.  Given:
* tiledbsoma/somacore support more functionality than is exposed here (e.g., varp_layers, etc)
* Census doesn't currently use these, due to _schema_ definition (not code)
* we want to clearly separate the arg names for clarity (the previous conversation where we decided to remove `varm_layers`)

I think we should land roughly here, to keep the code & schema modular:
* this function (`get_anndata`) should pass through all (or at least most) arguments supported by ExperimentAxisQuery.to_anndata.  That makes it future-proof and decoupled from the schema
* the newly added args should have clearly separated names (the current names are not clear per above comment)
* the error checking needs to detect collisions because AnnData only has one "dict" to shove them all in, and they might conflict.

Boiling this down, I suggest:
1. Add (and pass to query.to_anndata) the args `obsm_layers`, `varm_layers`, `obsp_layers`, and `varp_layers`.  These are just pass through.
2. Rename `add_obs_embeddings` and `add_var_embeddings` to something clear (see above discussion)
3. Do the error checks for collisions, and do them before you do any data loading.  Example below.

Example error check for obs:
```
if set(obsm_layers) & set(add_obs_embeddings):
    ... there is a collision error ...

....do same for varm ...
```

Reply:
Sounds good.